### PR TITLE
Add nix module to Salt

### PR DESCRIPTION
**Pull request by KaiSforza**
### What does this PR do?
This module adds the ability to work with nix packages, and do some
basic operations on the nix store. There is still some work to be done,
but this lets you orchestrate some basic tasks for a system with nix
installed.

### What issues does this PR fix or reference?
None, this is a new feature.

### Tests written?
No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

### CODE REVIEW

File: doc/ref/modules/all/salt.modules.nix.rst [hunk 1]
```diff
@@ -0,0 +1,6 @@
+================
+salt.modules.nix
+================
+
+.. automodule:: salt.modules.nix
+    :members:%
```
Comment (terminalmage) [inline doc/ref/modules/all/salt.modules.nix.rst]: I don't think that there is supposed to be a ``%`` here.

**Review by terminalmage — CHANGES_REQUESTED** [review]
This looks straightforward. Just one small issue in the RST file for the docs page.

**Review by cachedout — APPROVED** [review]

File: doc/ref/modules/all/salt.modules.nix.rst (hunk 1 above)
Comment (KaiSforza) [inline doc/ref/modules/all/salt.modules.nix.rst]: You are correct, that's my bad.

**Comment by KaiSforza** [comment]
@terminalmage Thanks, I removed that typo.

Also fixed a couple of problems I saw previously, as well as compacted the install output to be a single list, not a list of lists.

**Review by terminalmage — APPROVED** [review]

**Pull request merged**